backend/agent-manager/src/agent_manager/agent_manager.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from pathlib import Path
+from llama_stack_client import LlamaStackClient
+from llama_stack_client.types.shared_params.agent_config import AgentConfig
+from agent_manager.util import load_yaml
+from agent_manager.agent import Agent
+
+class AgentManager:
+    def __init__(self):
+        self._config = load_yaml(Path(__file__).parent / "config.yaml")
+        self._client = None
+        self._agents = []
+
+    def connect_to_llama_stack(self):
+        if self._client is None:
+            logging.debug("Connecting to LlamaStack")
+            self._client = LlamaStackClient(
+                base_url=self._config['llama_stack_url'],
+                timeout=self._config['timeout'],
+            )
+        else:
+            logging.debug("Already connected to LlamaStack")
+
+    def create_agents(self):
+        if self._client is None:
+            self.connect_to_llama_stack()
+        if self._agents == []:
+            logging.debug("Creating agents")
+            for agent in self._config["agents"]:
+                self.create_agent(agent)
+        else:
+            logging.debug("Agents already created")
+
+    def create_agent(self, agent: dict):
+        agent_config:AgentConfig={
+            "model": agent['model'],
+            "instructions": agent['instructions'],
+            "tool_choice": agent['tool_choice'],
```

Comment:
I think we need  "toolgroups":  as well here.

Reply:
As an example 

```python
            "toolgroups": [
                {
                    "name": "builtin::rag/knowledge_search",
                    "args": {"vector_db_ids": ["laptop-refresh-knowledge-base"]},
                },
                "mcp::asset_database",
                "mcp::servicenow",
            ],
```

We at least need to be able to specify 1 or more mcp servers and 1 or more knowledge bases.

In our configuration file we need two types of entries one for mcp server and a different one for knowlege bases. The mcp servers will map to entries like the last two toolgroups in the example above, and each knowledge base would make to an entry like there is for "laptop-refresh-knowledge-base" in the example

Reply:
The PRD has some more specific details of what should be in the config -https://docs.google.com/document/d/1Vgxgp77H7hW2VpUVO43d2vGybbfRpPnR3ZagSsli7mg/edit?tab=t.0#heading=h.r8703jdria1q

It's ok to add/modify those, but we should keep the two in sync (PRD/Implementation)